src/Plugin/Notifier/Sms.php

```diff
@@ -2,33 +2,114 @@
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Logger\LoggerChannelInterface $logger
+   *   The message_notify logger channel.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   The rendering service.
+   * @param \Drupal\message\MessageInterface $message
+   *   (optional) The message entity. This is required when sending or
+   *   delivering a notification. If not passed to the constructor, use
+   *   ::setMessage().
+   * @param \Drupal\sms\Provider\PhoneNumberProviderInterface $phone_number_provider
+   *   The sms service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, LoggerChannelInterface $logger, EntityTypeManagerInterface $entity_type_manager, RendererInterface $renderer, MessageInterface $message = NULL, PhoneNumberProviderInterface $phone_number_provider) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $logger, $entity_type_manager, $renderer, $message);
+
+    $this->phoneNumberProvider = $phone_number_provider;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition, MessageInterface $message = NULL) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('logger.channel.message_notify'),
+      $container->get('entity_type.manager'),
+      $container->get('renderer'),
+      $message,
+      $container->get('sms.phone_number')
+    );
+  }
+
   /**
    * {@inheritdoc}
    */
   public function deliver(array $output = []) {
-    throw new MessageNotifyException('This functionality depends on the SMS Framework module. See: https://www.drupal.org/node/2582937');
-    if (empty($this->message->smsNumber)) {
-      // Try to get the SMS number from the account.
-      $account = $this->message->uid->entitiy;
-      if (!empty($account->sms_user['number'])) {
-        $this->message->smsNumber = $account->sms_user['number'];
-      }
-    }
+    $message = trim($output['sms_message']);
+    $user = $this->message->getOwner();
+    $username = $user->getAccountName();
 
-    if (empty($this->message->smsNumber)) {
-      throw new MessageNotifyException('Message cannot be sent using SMS as the "smsNumber" property is missing from the Message entity or user entity.');
+    // Don't even bother sending an empty SMS. Waste of monies.
+    if ($message === '') {
+      $this->logger->error('Failed to send SMS to {username}. Message empty.', [
+        '$username' => $username,
+      ]);
+      return FALSE;
     }
 
-    return sms_send($this->message->smsNumber, strip_tags($output['message_notify_sms_body']));
+    try {
+      $result = $this->phoneNumberProvider->sendMessage($user, new SmsMessage(NULL, [], $message));
```

Comment:
sendMessage doesnt have any return value. However it does throw various exceptions.

Reply:
Thanks for using this service.

Reply:
This must be a bug on `sms` then. `PhoneNumberProvider` implements `PhoneNumberProviderInterface`. `PhoneNumberProviderInterface` defines `sendMessage` as:

```
  /**
   * Sends an SMS to an entity.
   *
   * @param \Drupal\Core\Entity\EntityInterface $entity
   *   The entity to send an SMS.
   * @param \Drupal\sms\Message\SmsMessageInterface $sms_message
   *   The SMS message to send to the entity.
   *
   * @return \Drupal\sms\Message\SmsMessageResultInterface|false
   *   The message result from the gateway.
   *
   * @throws \Drupal\sms\Exception\NoPhoneNumberException
   *   Thrown if entity does not have a phone number.
   */
  public function sendMessage(EntityInterface $entity, SmsMessageInterface $sms_message);
```

Also, what does this document refer to? https://www.drupal.org/docs/8/modules/sms-framework/integrating-with-sms-framework-0/interpreting-sms-framework-codes

Reply:
Looks like the docs are wrong. There is no return value for message, by design.

The queueing mechanism never returns status codes because the message is added a queue. Status is not known until later.